controllers/windowsmachine/windowsmachine_controller.go

```diff
@@ -116,10 +116,10 @@ func add(mgr manager.Manager, r reconcile.Reconciler) error {
 		// We need the create event to account for Machines that are in provisioned state but were created
 		// before WMCO started running
 		CreateFunc: func(e event.CreateEvent) bool {
-			return isWindowsMachine(e.Meta.GetLabels())
+			return isValidMachine(e.Object.DeepCopyObject()) && isWindowsMachine(e.Meta.GetLabels())
```

Comment:
Sorry for the basic question but why does the object need to be DeepCopied? Why can't you just use it directly?

Reply:
I wanted to be overly cautious. Might not actually be needed here since the func shouldn't mutate the object

Reply:
I'd rather not do it if its not necessary, since the predicates will be called on every machine object change. I'm not sure if its needed either